jedi/api/environment.py

```diff
@@ -149,7 +149,11 @@ def get_sys_path(self):
 def _get_virtual_env_from_var():
     var = os.environ.get('VIRTUAL_ENV')
     if var is not None:
-        if var == sys.prefix:
+        # Under macOS in some cases - notably when using Pipenv - the
+        # sys.prefix of the virtualenv is /path/to/env/bin/.. instead of
+        # /path/to/env so we need to fully resolve the paths in order to
+        # compare them.
+        if os.path.realpath(var) == os.path.realpath(sys.prefix):
```

Comment:
Any idea why that happens?
Is it related to certain Python versions maybe?

Reply:
And could you investigate why the fallback of `create_environment` below fails for you?
Is it because of an unsafe env (https://github.com/davidhalter/jedi/pull/1199), or something else?

Reply:
`VIRTUAL_ENV` is set in the configuration file as follows:
```
VIRTUAL_ENV="/Users/yasushi/.local/share/virtualenvs/venvtest-BgwP3k5I"
```
But `sys.prefix' is:
```
% python -c 'import sys; print(sys.prefix)'
/Users/yasushi/.local/share/virtualenvs/venvtest-BgwP3k5I/bin/..
```

I think that some kind of normalization is necessary like this pull request.

Reply:
How did you create the virtualenv?
What is "configuration file"?
What is `which python` / `command -v python`?
What is `python -V``